### DISCUSSION TITLE
Custom app Icon on Android to prevent shoulder-surfing attacks

### DISCUSSION THREAD

**Discussion by I-I-IT** [mullvad/mullvadvpn-app#8930] [Feature requests & Ideas]
### I have checked if others have suggested this already

- [X] I have checked this issue tracker to see if others have reported similar issues.

### Feature description

Rationale: Mullvad VPN is widely recognized as a trustworthy service, particularly effective at circumventing advanced online censorship, such as China's Great Firewall (GFW). However, a new threat has emerged for VPN users: real-life phone checks conducted by authorities. Reports indicate that in countries like China, police are actively searching phones for "malicious" or "scam" apps, including VPNs. Users found with such apps may face serious consequences [source1](https://www.theverge.com/2022/11/28/23482339/chinese-police-searching-phones-instagram-twitter-telegram-protests-covid) [source2](https://github.com/net4people/bbs/issues/354) .

This is why, Mullvad should implement a custom app icon option. It would allow the app to disguise itself as a random app, such as a weather app or a note taking app. This has already been implemented in [Proton VPN](https://protonvpn.com/support/disguise-app-icon).

A big limitation is that despite the app icon change, notifications of the Mullvad VPN tunnel status would still appear. It would be super nice if, instead of just sending to send notifications, it would send "crypted" message. Eg : The weather is good today == Connected, "You shouldn't go out now"== Disconnected.

Another limitation is that the original app name still appears in the settings. This mean it woudn't help against intensive checks by a cop, but would probably help against quick checks, or simply shoulder-surfing attempts.

### Alternative solutions

There is no other way to hide the app.

### Type of feature

- [ ] Better privacy/anonymity
- [X] Better at circumventing censorship
- [ ] Easier to use
- [ ] Other

### Operating System

- [X] Android
- [ ] iOS
- [ ] Windows
- [ ] macOS
- [ ] Linux

**Comment by Pururun** [comment]
Thanks for your feature requests.

We are looking into this, but unfortunately I can not give more information if/when this will be implemented.

**Comment by I-I-IT** [comment]
Good to know thank you.

**Comment by Rawa** [comment]
This has now been implemented in the app and released with `2026.2`, it is soon fully rolled out on Google Play Store as well. Currently the setting can be found under `Settings -> Appearance`.
